core/src/mina_polling_service.rs

```diff
@@ -213,14 +265,14 @@ pub async fn query_root(rpc_url: &str, length: usize) -> Result<StateHashAsDecim
         .best_chain
         .ok_or("Missing best chain field".to_string())?;
     let root = best_chain.first().ok_or("No root state")?;
-    Ok(root.state_hash_field.clone())
+    Ok(root.state_hash.clone())
 }
 
 pub async fn query_merkle(
     rpc_url: &str,
     state_hash: &str,
     public_key: &str,
-) -> Result<(Fp, Fp, Vec<MerklePath>, [u8; 32]), String> {
+) -> Result<(LedgerHash, Fp, Vec<MerklePath>, [u8; 32]), String> {
```

Comment:
This function now returns a `LedgerHash` instead of a `Fp` but when it's called, the returned `LedgerHash` is then converted to an `Fp`, Why changing the function signature?

Reply:
You are right! although this is consistent with the rest of the query functions that also return the OpenMina types. 

In the PR I'm working on currently about account inclusion we're going to have structs for defining a Mina Proof of Account the same way this PR introduces for a state proof, and in that case we would need the `LedgerHash` instead of the `Fp`. If you agree we can leave this as it is now as to not modify it again in the next PR.

Reply:
Makes sense 👍